dot/state/storage.go

```diff
@@ -203,6 +203,15 @@ func (s *StorageState) GetStorageByBlockHash(bhash common.Hash, key []byte) ([]b
 	return s.GetStorage(&header.StateRoot, key)
 }
 
+// GetStateRootFromBlock returns the state root hash of a given block hash
+func (s *StorageState) GetStateRootFromBlock(bhash common.Hash) (*common.Hash, error) {
+	header, err := s.blockState.GetHeader(bhash)
+	if err != nil {
+		return nil, err
+	}
+	return &header.StateRoot, nil
```

Comment:
is there a reason it returns a pointer?

Reply:
blockState.GetHeader() above returns a pointer and we later on user this as pointer in other functions.

Reply:
`blockState.GetHeader()` above returns a pointer, and we use it as a pointer in other functions later on.